bot/resources/tags/docstring.md

```diff
@@ -1,18 +1,16 @@
-A [`docstring`](https://docs.python.org/3/glossary.html#term-docstring) is a string with triple quotes that often used in file, classes, functions, etc. A docstring should have a clear explanation of exactly what the function does. You can also include descriptions of the function's parameter(s) and its return type, as shown below.
+A [`docstring`](https://docs.python.org/3/glossary.html#term-docstring) is a string with triple quotes that's placed at the top of files, classes and functions. A docstring should contain a clear explanation of what it's describing. You can also include descriptions of the subjects's parameter(s) and its return type, as shown below.
```

Comment:
> A docstring should contain a clear explanation of what it's describing.

Huh?

Reply:
We didn't want to enumerate files, classes and functions again, so we went with this instead, not sure how to improve this.

Reply:
What about..
> A docstring should contain a clear explanation of what it does and its purpose.

Reply:
it would need to be referring to the subject here, rather than the docstring itself. So what about `A docstring should contain a clear explanation of what it's describing does and its purpose.`

Reply:
`what it's describing does` still rubs me the wrong way.. what if we change that to `whatever`?

Or I guess completely different more like `A docstring should be a clear explanation of the purpose of what it describes.`

Reply:
I think something along the lines of `A docstring should describe the behaviour and the interface of what it's attached to` would fit.

Reply:
I like `what it's attached to`, but a docstring should not explain the interface. That is up to the function signature to do.

Reply:
With the interface I meant describing what some of the more complex parameters mean, not just how the signature looks- - maybe the original wording is clearer there